### ISSUE TITLE
Add CMORizing obs time series in devel

### ISSUE THREAD

**Issue by gleckler1** [PCMDI/pcmdi_metrics#317]
This code will be what is used to process (with CMOR) existing and newly acquired observational datasets.   To be jointly developed by PG and PD

**Comment by gleckler1** [comment]
@durack1 I thought what you had in ~git/pcmdi_metrics/src/python/pcmdi/scripts was a script to process monthly mean obs, but it (make_obs_clim.py) is for producing clims.   Do you have something we should start with for processing all obs from /work/climobs/orig to well structured CF compliant data, or do we need to start from scratch?   We can establish a framework for process all obs and then use cmor for it.

**Comment by durack1** [comment]
@gleckler1 good question. What I have is numerous obs-specific scripts to convert downloaded versions of data into CF-like gridded timeseries - so for Ishii, Aquarius etc

These are all in `ocean:/work/durack1/Shared/obs_data` and are very specific to the source data, which has come in `grib`, `hdf`, `netcdf` and other weird formats.

**Comment by gleckler1** [comment]
@durack1 I don't think there is an easy path here, but the hope is that we could modularize some (albeit limited) aspect of the processing, particularly since we plan to use cmor.   Maybe if we have obs specific functions in a module which one way or another read in the data and then do any manipulation necessary (rename variable, change units, reverse axis, etc)...  then have a driver script that can process some or all obs...

**Comment by durack1** [comment]
@gleckler1 I think the best thing we can hope for with limited time, is that each of the steps (download, rewrite etc) are logged so we at least have a trail to follow if questions arise.

We can then use standard scripts/cmor libraries to read and generate the climatologies and timeseries-style inputs for the package.. I think making things transparent is the first step, generalizing code would be a year of work..

**Comment by gleckler1** [comment]
@durack1  I think you are right about the required effort.  Part of keeping it transparent is keeping the original data and processing scripts in a common location and we have that with /work/clim_obs .... also, fortunately the obs4MIPs and ana4MIPs data do not need reprocessing.

**Comment by durack1** [comment]
@gleckler1 exactly, leveraging obs4mips data as a primary source is the best (quickest and simplest) path forward.. Even better is to advocate for data providers to submit their data into obs4mips..

**Comment by gleckler1** [comment]
@durack1 I've just re-discovered that all of the time series YZ had processed was done with ezcmor... not just clims....  this could help jumpstart things... if only there was more time...

**Comment by gleckler1** [comment]
@durack1 But we can (need to) make all obs clims consistent and right now the atm and ocn file names are not (below)...  so, like the model data we need a single code for processing obs clims... this can be a close variant of model code.

[gleckler1@oceanonly mo]$ pwd
/clim_obs/obs/ocn/mo
[gleckler1@oceanonly mo]$ ls */*/ac/*.nc
sos/JPL-Aquarius-v2/ac/sos_pcmdi-metrics_Omon_JPL-Aquarius-v2_201108-201304-clim.nc
sos/NODC-WOA09/ac/sos_pcmdi-metrics_Omon_NODC-WOA09_177201-200812-clim.nc
sos/UCSD-ARGO/ac/sos_pcmdi-metrics_Omon_UCSD-ARGO_200401-201212-clim.nc
tos/NOAA-OISST-v2/ac/tos_pcmdi-metrics_Omon_NOAA-OISST-v2_198202-201201-clim.nc
tos/NODC-WOA09/ac/tos_pcmdi-metrics_Omon_NODC-WOA09_177201-200812-clim.nc
tos/REMSS-AMSRE-L3-v7/ac/tos_pcmdi-metrics_Omon_REMSS-AMSRE-L3-v7_200301-201012-clim.nc
tos/UCSD-ARGO/ac/tos_pcmdi-metrics_Omon_UCSD-ARGO_200401-201212-clim.nc
tos/UKMETOFFICE-HadISST-v1-1/ac/tos_pcmdi-metrics_Omon_UKMETOFFICE-HadISST-v1-1_198002-200501-clim.nc
zos/CNES-AVISO-L4/ac/zos_pcmdi-metrics_Omon_CNES-AVISO-L4_199301-201012-clim.nc

[gleckler1@oceanonly mo]$ pwd
/clim_obs/obs/atm/mo
[gleckler1@oceanonly mo]$ ls -lrt */*/ac/*.nc
-rwxrwxr-x 1 nobody climate 11618316 Jul 12  2011 va/ERA40/ac/va_ERA40_000001-000012_ac.nc
-rwxrwxr-x 1 nobody climate 46121292 Jul 12  2011 va/JRA25/ac/va_JRA25_000001-000012_ac.nc
-rwxrwxr-x 1 nobody climate 23713116 Jul 12  2011 va/ERAINT/ac/va_ERAINT_000001-000012_ac.nc
-rwxrwxr-x 1 nobody climate  1410128 Jul 12  2011 ts/ERAINT/ac/ts_ERAINT_000001-000012_ac.nc
-rwxrwxr-x 1 nobody climate  1410096 Jul 12  2011 hfss/ERAINT/ac/hfss_ERAINT_000001-000012_ac.nc
-rwxrwxr-x 1 nobody climate   510460 Jul 12  2011 pr/GPCP/ac/pr_GPCP_000001-000012_ac.nc
-rwxrwxr-x 1 nobody climate   517336 Jul 12  2011 psl/ERA40/ac/psl_ERA40_000001-000012_ac.nc
-rwxrwxr-x 1 nobody climate  1410136 Jul 12  2011 psl/ERAINT/ac/psl_ERAINT_000001-000012_ac.nc
-rwxrwxr-x 1 nobody climate 46121296 Jul 12  2011 zg/JRA25/ac/zg_JRA25_000001-000012_ac.nc
-rwxrwxr-x 1 nobody climate 23713120 Jul 12  2011 zg/ERAINT/ac/zg_ERAINT_000001-000012_ac.nc
-rwxrwxr-x 1 nobody climate 11618320 Jul 12  2011 zg/ERA40/ac/zg_ERA40_000001-000012_ac.nc
-rwxrwxr-x 1 nobody climate   517624 Jul 12  2011 tas/ERA40/ac/tas_ERA40_000001-000012_ac.nc
-rwxrwxr-x 1 nobody climate  2022712 Jul 12  2011 tas/JRA25/ac/tas_JRA25_000001-000012_ac.nc
-rwxrwxr-x 1 nobody climate  1410424 Jul 12  2011 tas/ERAINT/ac/tas_ERAINT_000001-000012_ac.nc
-rwxrwxr-x 1 nobody climate  1410096 Jul 12  2011 tauv/ERAINT/ac/tauv_ERAINT_000001-000012_ac.nc
-rwxrwxr-x 1 nobody climate 23713248 Jul 12  2011 hur/ERAINT/ac/hur_ERAINT_000001-000012_ac.nc
-rwxrwxr-x 1 nobody climate 11618448 Jul 12  2011 hur/ERA40/ac/hur_ERA40_000001-000012_ac.nc
-rwxrwxr-x 1 nobody climate 46121288 Jul 12  2011 ta/JRA25/ac/ta_JRA25_000001-000012_ac.nc
-rwxrwxr-x 1 nobody climate 23713112 Jul 12  2011 ta/ERAINT/ac/ta_ERAINT_000001-000012_ac.nc
-rwxrwxr-x 1 nobody climate 11618312 Jul 12  2011 ta/ERA40/ac/ta_ERA40_000001-000012_ac.nc
-rwxrwxr-x 1 nobody climate  2022724 Jul 12  2011 uas/JRA25/ac/uas_JRA25_000001-000012_ac.nc
-rwxrwxr-x 1 nobody climate   517636 Jul 12  2011 uas/ERA40/ac/uas_ERA40_000001-000012_ac.nc
-rwxrwxr-x 1 nobody climate  1410436 Jul 12  2011 uas/ERAINT/ac/uas_ERAINT_000001-000012_ac.nc
-rwxrwxr-x 1 nobody climate  1410096 Jul 12  2011 tauu/ERAINT/ac/tauu_ERAINT_000001-000012_ac.nc
-rwxrwxr-x 1 nobody climate  2022716 Jul 12  2011 huss/JRA25/ac/huss_JRA25_000001-000012_ac.nc
-rwxrwxr-x 1 nobody climate   510416 Jul 12  2011 prw/RSS/ac/prw_RSS_000001-000012_ac.nc
-rwxrwxr-x 1 nobody climate 23713116 Jul 12  2011 ua/ERAINT/ac/ua_ERAINT_000001-000012_ac.nc
-rwxrwxr-x 1 nobody climate 46121292 Jul 12  2011 ua/JRA25/ac/ua_JRA25_000001-000012_ac.nc
-rwxrwxr-x 1 nobody climate 11618316 Jul 12  2011 ua/ERA40/ac/ua_ERA40_000001-000012_ac.nc
-rwxrwxr-x 1 nobody climate  1410436 Jul 12  2011 vas/ERAINT/ac/vas_ERAINT_000001-000012_ac.nc
-rwxrwxr-x 1 nobody climate  2022724 Jul 12  2011 vas/JRA25/ac/vas_JRA25_000001-000012_ac.nc
-rwxrwxr-x 1 nobody climate   517636 Jul 12  2011 vas/ERA40/ac/vas_ERA40_000001-000012_ac.nc
-rwxrwxr-x 1 nobody climate 24071928 Jul 13  2011 hus/JRA25/ac/hus_JRA25_000001-000012_ac.nc
-rwxrwxr-x 1 nobody climate 23713120 Jul 13  2011 hus/ERAINT/ac/hus_ERAINT_000001-000012_ac.nc
-rwxrwxr-x 1 nobody climate 11618320 Jul 13  2011 hus/ERA40/ac/hus_ERA40_000001-000012_ac.nc
-rwxrwxr-x 1 nobody climate  6232932 Jul 13  2011 rswcrf/CERES/ac/rswcrf_CERES_000001-000012_ac.nc
-rwxrwxr-x 1 nobody climate  6232932 Jul 13  2011 rlwcrf/CERES/ac/rlwcrf_CERES_000001-000012_ac.nc
-rw-rw-r-- 1 nobody climate 27699772 May 28  2013 pr/TRMM/ac/pr_TRMM_000001-000012_ac.nc
-rw-rw-r-- 1 nobody climate 52897576 May 28  2013 hus/AIRS/ac/hus_AIRS_000001-000012_ac.nc
-rw-rw-r-- 1 nobody climate  7165672 May 28  2013 hus/MLS/ac/hus_MLS_000001-000012_ac.nc
-rw-rw-r-- 1 nobody climate 52897568 May 28  2013 ta/AIRS/ac/ta_AIRS_000001-000012_ac.nc
-rw-rw-r-- 1 nobody climate  7165664 May 28  2013 ta/MLS/ac/ta_MLS_000001-000012_ac.nc
-rw-rw-r-- 1 nobody climate  3130932 May 28  2013 tos_old/AMSRE/ac/tos_AMSRE_000001-000012_ac.nc
-rw-rw-r-- 1 nobody climate  3130860 May 29  2013 zos_old/AVISO/ac/zos_AVISO_000001-000012_ac.nc
-rw-rw-r-- 1 nobody climate  3130916 May 29  2013 rsut/CERES/ac/rsut_CERES_000001-000012_ac.nc
-rw-rw-r-- 1 nobody climate  3130896 May 29  2013 rsdt/CERES/ac/rsdt_CERES_000001-000012_ac.nc
-rw-rw-r-- 1 nobody climate  3130956 May 29  2013 rlut/CERES/ac/rlut_CERES_000001-000012_ac.nc
-rw-rw-r-- 1 nobody climate  3130896 May 29  2013 rlutcs/CERES/ac/rlutcs_CERES_000001-000012_ac.nc
-rw-rw-r-- 1 nobody climate  3130896 May 29  2013 rsutcs/CERES/ac/rsutcs_CERES_000001-000012_ac.nc
-rw-rw-r-- 1 nobody climate  3130960 May 29  2013 clt/MODIS/ac/clt_MODIS_000001-000012_ac.nc

**Comment by gleckler1** [comment]
@durack1 I've just realized that although the obs ocn clims have been standardized in general the month mean ocean time series are not.   That has not come up until now as we are getting variability analysis together which will rely on multiple monthly mean sst datasets.   Lets chat about our strategy...

**Comment by doutriaux1** [comment]
guys this seems to be way off topic from the PMP, I would recommend creating a separate repo for these sort of things. pcmdi_metrics_maintain or something like that, we do not want to pollute this repo with things that are not needed at all by the end user.

**Comment by durack1** [comment]
@doutriaux1 we have some demo files at https://github.com/PCMDI/pcmdi_metrics/tree/master/demo - I think it's best we keep code together, but not anything else.. Creating a separate repo at this stage is a little pre-emptive

**Comment by doutriaux1** [comment]
yes but it's 5mn now vs 2 days and a really really horrible repo in a few months :wink:

**Comment by durack1** [comment]
@doutriaux1 point taken.. Ok will have to think about organizing another repo..

**Comment by gleckler1** [comment]
@durack1 finally making some progress!
